components/butterfly/src/rumor/dat_file.rs

```diff
@@ -233,9 +234,12 @@ impl DatFile {
 
     pub fn write(&self, server: &Server) -> Result<usize> {
         let mut header = Header::default();
-        let tmp_path = self
-            .path
-            .with_extension(thread_rng().gen_ascii_chars().take(8).collect::<String>());
+        let tmp_path = self.path.with_extension(
+            thread_rng()
+                .sample_iter(&Alphanumeric)
+                .take(8)
```

Comment:
As long as we're making a change here, can we avoid the magic number? Or perhaps replace this with something like [`mktemp`](https://crates.io/crates/mktemp)? We're already using it elsewhere.

Reply:
For example: https://github.com/habitat-sh/habitat/pull/5511

Reply:
I see you've already addressed this in #5511

Reply:
Same idea, different place in the code. I'm holding back on merging that until after the release since it's not urgent.